### PR TITLE
Feature/create project

### PR DESCRIPTION
**Pull request by eduskx**
note: styling will be refactored and dynamic input fields will be merged into one component in future user stories.

### CODE REVIEW

File: components/Form/DynamicArrayInput.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+      <StyledLabel htmlFor={label}>{label}</StyledLabel>
+      {state.map((element, index) => (
+        <StyledMaterialsWrapper key={index}>
+          <StyledInput
+            required
+            id={label}
+            value={element}
+            onChange={(event) => handleChange(index, event)}
+            type="text"
+          />
```
Comment (stepris) [inline components/Form/DynamicArrayInput.js:36]: Great use of React's controlled component pattern!

File: components/Form/DynamicStepsInput.js [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            required
+            id={steps}
+            value={step.description}
+            onChange={(event) => handleChange(index, event)}
+            type="text"
+          />
+          <StyledDeleteButton
+            type="button"
+            onClick={() => handleRemoveField(index)}
+          >
```
Comment (stepris) [inline components/Form/DynamicStepsInput.js:35]: The steps do not always have a consecutive sequence of numbers when steps are deleted during creation

**Review by stepris — COMMENTED** [review]
Fantastic work implementing this rather large feature! 

There's definitely room for adding more error handling, validation, and accessibility considerations. The idea of combining ArrayInput and StepsInput into a general component sounds excellent. 

However, these might be things for future user stories. 

Keep up the good work, very nicely done!

File: components/Form/DynamicStepsInput.js (hunk 2 above)
Comment (eduskx) [reply components/Form/DynamicStepsInput.js:35]: Thanks! We would've probably never noticed haha

File: components/Form/index.js [hunk 3]
```diff
@@ -14,9 +14,11 @@ export default function Form({ setNewProjects, projects }) {
 
     const formData = new FormData(event.target);
     const newProject = Object.fromEntries(formData);
-    newProject.id = `${projects.length + 1}`;
+    const idOfLatestProject = Number(projects[projects.length - 1].id);
+    newProject.id = `${idOfLatestProject + 1}`;
```
Comment (stepris) [inline components/Form/index.js:18]: At a first glance, this seems like a good solution for ensuring that IDs are generated continuously from the last project ID 🙌. Nice! 
This should prevent the potential issue of assigning the same ID again when projects are deleted from the array. It requires that the projects array is not empty, though. Here's a potential solution:

`const idOfLatestProject = Number(projects[projects.length - 1]?.id || 0);`

**Pull request merged**